redbot/core/commands/converters.py

```diff
@@ -0,0 +1,80 @@
+import re
+from datetime import timedelta
+from typing import TYPE_CHECKING
+
+from redbot.core.commands import Converter, BadArgument
+
+if TYPE_CHECKING:
+    from .commands import Context
+
+__all__ = ["TimedeltaConverter"]
+
+TIME_RE = re.compile(
+    r"((?P<days>\d+?)\s?(d(ays?)?))?\s?((?P<hours>\d+?)\s?(hours?|hrs|hr?))?\s?((?P<minutes>\d+?)\s?(minutes?|mins?|m))?\s?((?P<seconds>\d+?)\s?(seconds?|secs?|s))?\s?",
+    re.I,
+)
+
+
+class TimedeltaConverter(Converter):
+    """
+    Convert to a :class:`datetime.timedelta` class.
+
+    The converter supports seconds, minutes, hours and days.
+
+    Time can be attached to the unit (only use the first letter) or not
+    (then use the full word).
+
+    .. admonition:: Example
+
+        *   ``50s`` = ``50 seconds`` -> :class:`datetime.timedelta(seconds=50)`
+        *   ``12m`` = ``12 minutes`` -> :class:`datetime.timedelta(seconds=720)`
+        *   ``4h`` = ``4 hours`` -> :class:`datetime.timedelta(seconds=14400)`
+        *   ``1d`` = ``1 day`` -> :class:`datetime.timedelta(days=1)`
+
+        Using the converter with a command:
+
+        .. code-block:: python3
+
+            @commands.command()
+            async def timer(self, ctx, time: commands.TimedeltaConverter):
+                await self.start_timer(time)
+
+        Using the converter manually:
+
+        .. code-block:: python3
+
+            async def convert_time(ctx: Context, text: str) -> datetime.timedelta:
+                time = await commands.TimedeltaConverter().convert(ctx, text)
+                return time
+    """
+
+    async def convert(self, ctx: "Context", argument: str) -> timedelta:
```

Comment:
I'm not sure what ctx is doing here. It's not used for the rest of the code in this function, and the "type" checking is just "Context". Did you mean to check for `commands.Context` ?

Reply:
I tried to follow what the `discord.py` docs recommand. Please read this, and if you still want to remove that argument, tell me.
https://discordpy.readthedocs.io/en/rewrite/ext/commands/commands.html#advanced-converters

Reply:
the discord.py docs recommend using basic converters unless you require ctx. Since you don't require ctx, then a basic converter will likely be a better approach.
https://discordpy.readthedocs.io/en/rewrite/ext/commands/commands.html#basic-converters

Reply:
I find it easier for the users to use the advanced converters, because the core discord.py's converters are advanced. The users are used to call a function like that:
```py
await commands.MemberConverter.convert(ctx, text)
```
Using the basic converters would give this:
```py
commands.timedelta_converter(text)
```
Using a different style is quite a mess, especially since **`redbot.core.commands` inherits from `discord.ext.commands`** and include their converters. We should just keep the same style as the other converters. Just imagine:
- For a member, I call `commands.MemberConverter.convert()`
- For a channel, I call `commands.TextChannelConverter.convert()`
- For a role, I call `commands.RoleConverter.convert()`
- For time, I call `commands.timedelta_converter()`

That doesn't match at all with the other converters and will confuse the users. That's why I'm using advanced converters, so it works the same as the other ones, which are often used.

Basic converters may be useful for a personnal cog, but not for an API that already uses an other type of converters under the same module!

Reply:
discord.py includes basic converters as well, ex `clean_content`. More than that, basic converters are all over the bot, just look at any `int` or `str` typehints. Those are basic converters in d.py's eyes as well. 

The advanced converters are advanced *because* they require `ctx`. The advanced converters require `ctx` because they are getting data about the context to return Discord objects. Your converter does not require `ctx` because it's neither returning any Discord objects nor getting any data about Discord's context.

Creating a convoluted method to convert to match Discord converters which *require* that convoluted method to function is quite simply absurd. You should match the functionality of what you require, keep it as simple as possible, and above all actually match how d.py functions with its typehint conversions rather than make nonsense excuses for what you coded.

Besides which, what happens if I want to convert an argument to a timedelta in a Discord event? Your method *requires* me to have a Context object when it's not needed, so I'd have to somehow find or create one or be SOL. I could pass `None`, but what looks better and is easier to remember in your eyes?

`commands.timedelta_converter(my_string)`
`await commands.TimedeltaConverter().convert(None, my_string)`

Reply:
Actually, you're right. There's nothing used here which would need a coro or ctx, so This can easily just be a basic converter.